python/lsst/pipe/tasks/subtractBrightStars.py

```diff
@@ -0,0 +1,325 @@
+            # compute scaling factor
+            XYSum = afwMath.makeStatistics(XY, self.statsFlag, self.statsControl).getValue()
+            XXSum = afwMath.makeStatistics(XX, self.statsFlag, self.statsControl).getValue()
+            scalingFactor = XYSum / XXSum if XXSum else 0
+        else:
+            raise AttributeError(f'Unknown scalingType "{self.config.scalingType}"; implemented options '
+                                 'are "annularFlux" and "leastSquare".')
+        if inPlace:
+            model.image *= scalingFactor
+        return scalingFactor
+
+    def run(self, inputExposure, inputBrightStarStamps, inputExtendedPsf, skyCorr=None, dataId=None):
+        """Iterate over all bright stars in an exposure to scale the extended
+        PSF model before subtracting bright stars.
+
+        Parameters
+        ----------
+        inputExposure : `afwImage.exposure.exposure.ExposureF`
+            The image from which bright stars should be subtracted.
+        inputBrightStarStamps :
+                `lsst.meas.algorithms.brightStarStamps.BrightStarStamps`
+            Set of stamps centered on each bright star to be subtracted,
+            produced by running
+            `lsst.pipe.tasks.processBrightStars.ProcessBrightStarsTask`.
+        inputExtendedPsf : `lsst.pipe.tasks.extended_psf.ExtendedPsf`
+            Extended PSF model, produced by
+            `lsst.pipe.tasks.extended_psf.MeasureExtendedPsfTask`.
+        skyCorr : `lsst.afw.math.backgroundList.BackgroundList` or ``None``,
+                  optional
+            Full focal plane sky correction, obtained by running
+            `lsst.pipe.drivers.skyCorrection.SkyCorrectionTask`.
+        dataId : `dict` or `lsst.daf.butler.DataCoordinate`
+            The dataId of the exposure (and detector) bright stars should be
+            subtracted from.
+
+        Returns
+        -------
+        subtractorExp : `afwImage.ExposureF`
+            An Exposure containing every bright star's profile; its image can
+            then be subtracted from the input exposure.
+        invImages : `list` [`afwImage.MaskedImageF`]
+            A list of small images ("stamps") containing the model, each scaled
+            to its corresponding input bright star.
+        """
+        inputExpBBox = inputExposure.getBBox()
+        if self.config.doApplySkyCorr:
+            self.log.info("Applying sky correction to exposure %s (exposure will be modified in-place).",
+                          dataId)
+            self.applySkyCorr(inputExposure, skyCorr)
+        # Create an empty image the size of the exposure
+        # TODO: DM-31085 set mask planes
+        subtractorExp = afwImage.ExposureF(bbox=inputExposure.getBBox())
+        subtractor = subtractorExp.maskedImage
+        # make a copy of the input model
+        model = inputExtendedPsf(dataId["detector"]).clone()
+        modelStampSize = model.getDimensions()
+        inv90Rots = 4 - inputBrightStarStamps.nb90Rots
```

Comment:
I'm not sure if there's any check to ensure `nb90Rots` is an integer between 0-3. So this might be better off as 
```python
inv90Rots = 4 - inputBrightStarStamps.nb90Rots%4
```